src/pp_creators/basic_creator.py

```diff
@@ -317,6 +324,8 @@ def has_negligible_impedance(self, ce: ConductingEquipment) -> bool:
             return True
         if isinstance(ce, EquivalentBranch):
             return True
+        if isinstance(ce, EnergyConnection):
```

Comment:
As mentioned in the comment above, EnergyConnections are not meant to be collapsed into topological nodes. What is the issue you are trying to solve by collapsing them?

Reply:
I'm working on an LV version of the hosting capacity script in the examples repo. The way the script works is I sample a random selection of energy consumers to add loads/sgens to. I need a EC -> topological node mapping to do this.

Reply:
After giving it some thought, I think the best solution would be to get the bus from the connectivity node above the energy consumer.

Reply:
That's correct, that would be the approach. Each EC and PEC should have a terminal associated with a topological node. I I think you can use the mRID of the terminal in both the EC and PEC to get the topological node back from the mapping (the mapping code adds entries for the connectivity node mRID and all its terminals mRIDs back to the topological node if I remember correctly).